### ISSUE TITLE
`poetry run` swallows `--help` from the underlying sub-command

### ISSUE THREAD

**Issue by kevinastone** [sdispater/poetry#1215]
- [X] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [X] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [X] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: 
- **Poetry version**: 
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: 

## Issue

Poetry is swallowing the `--help` long-option and not passing it to the underlying command.  In this specific example, I was trying to get the help-text from `black`.

```
(.venv) kstone@kstone-mbp ~> poetry run black --help
USAGE
  poetry run <args1> ... [<argsN>]

ARGUMENTS
  <args>                 The command and arguments/options to run.

GLOBAL OPTIONS
  -h (--help)            Display this help message
  -q (--quiet)           Do not output any message
  -v (--verbose)         Increase the verbosity of messages: "-v" for normal output, "-vv" for more verbose output and
                         "-vvv" for debug
  -V (--version)         Display this application version
  --ansi                 Force ANSI output
  --no-ansi              Disable ANSI output
  -n (--no-interaction)  Do not ask any interactive question

(.venv) kstone@kstone-mbp ~> poetry run black --version
black, version 19.3b0
```

**Comment by kevinastone** [comment]
There doesn't seem to be a double-dash separator to prevent parsing the sub-command's arguments:

```
(.venv) kstone@kstone-mbp ~> poetry run -- black --help
USAGE
  poetry run <args1> ... [<argsN>]

ARGUMENTS
  <args>                 The command and arguments/options to run.

GLOBAL OPTIONS
  -h (--help)            Display this help message
  -q (--quiet)           Do not output any message
  -v (--verbose)         Increase the verbosity of messages: "-v" for normal output, "-vv" for more verbose output and
                         "-vvv" for debug
  -V (--version)         Display this application version
  --ansi                 Force ANSI output
  --no-ansi              Disable ANSI output
  -n (--no-interaction)  Do not ask any interactive question
```

**Comment by kevinastone** [comment]
The issue is in `CliKit` since it doesn't respect it's own `--` stop-token when processing the help option:

Help special case:
https://github.com/sdispater/clikit/blob/9b0e11e0863c9ed10bff743436ce405add0b6121/src/clikit/config/default_application_config.py

Arguments are tested here:
        while token:
            # "--" stops argument parsing
            if token == "--":
                break

**Comment by absassi** [comment]
See also #929

**Comment by jacebrowning** [comment]
@sdispater I think that fixing this bug should be part of the 1.0 milestone.

**Comment by kevinastone** [comment]
This should be fixed via c4b2c38575ff93e61044c2e67bf01a4fd03d38d0 since the double-dash was added in [clikit v0.4.0](https://github.com/sdispater/clikit/blob/master/CHANGELOG.md#fixed)
